src/utils/fundamentals/fundamentals.js

```diff
@@ -0,0 +1,149 @@
+import * as utils from '../network/utils'
+import { searchData } from '../network'
+
+/**
+ * Make request and search data
+ * @function countries
+ * @param {object} params - object of parameters:
+ * @param {string} params.code 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function countries({ code }, callback) {
+  const params = {
+    AMaaSClass: 'fundamentalCountries',
+    query: {countryCode: [code]}
+  }
+  //if code is not specified, retun all countries 
+  let promise = searchData(params).then(result => {
+    if (typeof callback === 'function') {
+      callback(null, result)
+    }
+    return result
+   })
+  if (typeof callback !== 'function') {
+    // return promise if callback is not provided
+    return promise
+  }
+  promise.catch(error => callback(error))
+}
+
+/**
+ * Make request and search data
+ * @function calcBusinessDate
+ * @param {object} params - object of parameters:
+ * @param {string} params.codes - A country code string or an array of country codes strings 
+ * @param {string} params.date - start date
+ * @param {string} params.offset
+ * @param {string} params.invalidDates - an invalid date string or an array of invalid dates strings 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function calcBusinessDate({date, codes, offset, invalidDates}, callback) //invalid dates
+{
+    if(!date){
```

Comment:
If the function returns here, the caller will get unpredictable results. Instead we have to take the appropriate error paths:

If `callback` is supplied, we just `return callback('Error: missing parameters. Date, codes and offset must be defined')`.

If callback is not supplied, we can return a Promise rejection directly.

```javascript
if (!date || !codes || !offset) {
  if (typeof callback === 'function') {
    return callback('Error: missing parameters etc.')
  } else {
    return Promise.reject('Error: missing parameters etc.')
  }
}
```